app/helpers/application_helper.rb

```diff
@@ -122,12 +122,9 @@ def sanitized_sidebar(text)
   def follow_button(followable, style = "full", classes = "")
     return if followable == DELETED_USER
 
-    tag :button, # Yikes
-        class: "crayons-btn follow-action-button #{classes} whitespace-nowrap",
-        data: {
-          :info => { id: followable.id, className: followable.class.name, style: style }.to_json,
-          "follow-action-button" => true
-        }
+    info   = { id: followable.id, className: followable.class.name, style: style }.to_json
+    button_tag("Follow", type: "button", data: { info: info },
+                         class: "crayons-btn follow-action-button #{classes} whitespace-nowrap")
```

Comment:
This doesn't exist in the helper methods as @aitchiss mentioned. The previous code resulted in: `<button properties></button>`

I suggestion you use the `tag` method like this:

```ruby
info = { id: followable.id, className: followable.class.name, style: style }
classes = "crayons-btn follow-action-button whitespace-nowrap #{classes}"
tag.button(name: :button, type: :button, data: { info: info }, class: classes)
```

see https://api.rubyonrails.org/classes/ActionView/Helpers/TagHelper.html#method-i-tag

ps. `to_json` isn't needed, they're already encoded as JSON:

> Data attribute values are encoded to JSON, with the exception of strings, symbols and BigDecimals. 
see

Reply:
Thanks a lot @rhymes 👍🏻, that's the e2e test I did not understand, but with the screenshot by @aitchiss I understand it breaks the editor. I've added your suggestion and the editor is ok. The HTML validator seems happy it. It produces `<button></button>` without any text, but it seems  it's added with JavaScript anyway:

<img width="751" alt="forem-follow-btn" src="https://user-images.githubusercontent.com/6878027/124770337-8bb62780-df3a-11eb-8d25-788291835dc9.png">